modules/meta-schema.json

```diff
@@ -182,5 +120,44 @@
             }
         }
     },
+    "definitions": {
+        "elementProperties": {
+            "type": "object",
+            "patternProperties": {
+                ".*": {
+                    "type": "object",
+                    "properties": {
+                        "type": {
+                            "type": "string",
+                            "description": "Type of the channel element",
+                            "enum": ["map", "file", "directory", "string", "integer", "float", "boolean", "list"]
```

Comment:
are these all nextflow types?

Reply:
Those are the types that we define, same as before :) The nextflow "type" I called it qualifier. 
Can be discussed if we need this or not, because it can be inferred from the `main.nf` or from the ontology (once we add it)

Reply:
I would prefer nextflow types, especially if we get type hinting in nextflow in the future. Or what do you think, @ewels ?

Reply:
I converted the PR to a draft because the tools PR is not ready anyway. And we can keep discussing this. 
I think @ewels was against both `type` and `qualifier` in the `meta.yml` 😄

Reply:
Hmm, at least one of them would make the move over to typed nextflow easier I think...

Reply:
Removed qualifier after [some discussion](https://github.com/nf-core/modules/pull/5867#discussion_r1656681444)